src/shared/install.c

```diff
@@ -2790,25 +2790,34 @@ static int do_unit_file_disable(
 
         _cleanup_(install_context_done) InstallContext ctx = { .scope = scope };
         _cleanup_set_free_free_ Set *remove_symlinks_to = NULL;
+        InstallInfo *info;
+        bool has_install_info = false;
         int r;
 
         STRV_FOREACH(name, names) {
-                if (!unit_name_is_valid(*name, UNIT_NAME_ANY))
-                        return install_changes_add(changes, n_changes, -EUCLEAN, *name, NULL);
```

Comment:
Why the check is dropped?

Reply:
I mistakenly believed that `install_info_traverse` would take care of this... Sorry.

Reply:
Previously if `install_info_add` fails, it simply returns `r` instead of calling `install_changes_add`. Does my change fit or should I stick to the original one?

Reply:
The change you made in the below looks good to me.